homeassistant/helpers/condition.py

```diff
@@ -1054,8 +951,10 @@ async def async_validate_condition_config(
         return config
 
     platform = await _async_get_condition_platform(hass, config)
-    if platform is not None and hasattr(platform, "async_validate_condition_config"):
-        return await platform.async_validate_condition_config(hass, config)
+    if platform is not None:
+        if hasattr(platform, "async_validate_condition_config"):
+            return await platform.async_validate_condition_config(hass, config)
+        return cast(ConfigType, platform.CONDITION_SCHEMA(config))
```

Comment:
instead of adding another way for the platform to provide validation, why not implement `async_validate_condition_config` in the sun condition platform?
otherwise there will be 2 ways of doing the same thing in a condition platform, if I am reading this correctly

Reply:
We have the same pattern for triggers, it's mostly because it's unnecessary to spawn a task to do some simple validation.

Reply:
got it! thanks.